assembly/src/parsers/u32_ops.rs

```diff
@@ -186,19 +191,74 @@ pub fn parse_u32xor(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Ass
     Ok(())
 }
 
-/// TODO: implement
-pub fn parse_u32not(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Translates u32not assembly instruction to VM operation PAD U32OR SWAP U32XOR.
+pub fn parse_u32not(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            span_ops.push(Operation::Pad);
+            span_ops.push(Operation::U32or);
+            span_ops.push(Operation::Swap);
+            span_ops.push(Operation::U32xor);
+        }
+        _ => return Err(AssemblyError::extra_param(op)),
+    }
+
+    Ok(())
 }
 
-/// TODO: implement
-pub fn parse_u32shl(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Translates u32shl.x assembly instruction to VM operation PUSH(2^x) MUL U32DIV DROP.
+pub fn parse_u32shl(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0..=1 => return Err(AssemblyError::missing_param(op)),
+        2 => {
+            // Assert the value is u32
+            span_ops.push(Operation::U32split);
+            span_ops.push(Operation::Not);
+            span_ops.push(Operation::Assert);
+
+            match op.parts()[1].parse::<u32>() {
```

Comment:
We should probably use `parse_int_param()` function from the parent module for argument parsing here.